### PR TITLE
Add FullStory Analytics Components

### PR DESCRIPTION
**Pull request by cbauerman**
#### Changes proposed in this Pull Request

* Add new Component, `<FullStoryRecorder/>` to Calypso
   *  Triggers tacking for logged in, US users, when rendered ( and `record` property not set to false )
* Add new Component, `<AnalyticsSafeContainer/>` to protect sensitive user information
   * In addition to FullStory this should implement safety for any other new analytics packages used
   * Care should be user to make all areas that could need protecting are found and covered
 

#### Testing instructions
 1. Add `FullStoryRecorder` and `AnalyticsSafeContainer` to multiple pages of a flow ( see my example at #52648 )
 2. Run in development mode
 3. Verify recording the flow works in the FullStory dashboard

### CODE REVIEW

**Comment by simison** [comment]
I remember @tyxla implemented some generic libraries to add these scripts when we started using [HotJar](https://github.com/Automattic/wp-calypso/search?q=hotjar). Perhaps this could go to that same library to avoid boilerplate?

**Comment by tyxla** [comment]
> I remember @tyxla implemented some generic libraries to add these scripts when we started using [HotJar](https://github.com/Automattic/wp-calypso/search?q=hotjar). Perhaps this could go to that same library to avoid boilerplate?

In case that helps, @simison is talking about this one: https://github.com/Automattic/wp-calypso/pull/30809

**Comment by razvanpapadopol** [comment]
Can we use FullStory also in wp-admin? This would involve running the script on `[site_slug].wordpress.com` sub-domains instead of on `wordpress.com`.
I'm asking in case we might want to use it for screens rendered inside the editor like Welcome tour, Page picker, Focused launch.
cc: @simison

In case the points above are true, it makes sense to add the script to `calypso-analytics` [package](https://github.com/Automattic/wp-calypso/tree/trunk/packages/calypso-analytics) and import it back in `client` and `apps/editing-toolkit`.

Without access to `client/state`, we can use the `User` store from [data-stores package](https://github.com/Automattic/wp-calypso/tree/trunk/packages/data-stores) to get the country code of current user. 
Typing `wp.data.select( 'automattic/user' ).getCurrentUser().user_ip_country_code` in the JS console when in https://wordpress.com/new should return the same data as `getCurrentUserCountryCode( state )` does in Calypso.

**Comment by simison** [comment]
Yep, we'll need this in wp-admin for both simple and Atomic sites, as well sites with custom domains (so not just `*.wordpress.com` subdomains). "Half of .com" is now on wp-admin. cc @obenland

File: client/lib/analytics/ad-tracking/constants.js [hunk 1]
```diff
@@ -69,6 +69,7 @@ export const TRACKING_IDS = {
 	bingInit: '4074038',
 	facebookInit: '823166884443641',
 	facebookJetpackInit: '919484458159593',
+	fullStoryOrgId: '1149DX',
```
Comment (jaswrks) [inline client/lib/analytics/ad-tracking/constants.js]: Should this be `120RG4` ?

**Comment by cbauerman** [comment]
⚠️ Branch re-based from trunk, local versions will need to be updated ⚠️

**Comment by cbauerman** [comment]
Closed in favor of #52768

**Comment by cbauerman** [comment]
closed in favor of #52768

**Pull request closed without merging**